src/legacy/core_plugins/metrics/server/plugin.ts

```diff
@@ -26,7 +26,8 @@ import { visDataRoutes } from './routes/vis';
 import { SearchStrategiesRegister } from './lib/search_strategies/search_strategies_register';
 
 export class Plugin {
-  public setup(core: InternalCoreSetup) {
+  // TODO: Remove 'any' as soon as CoreSetup is completed.
+  public setup(core: CoreSetup | any) {
```

Comment:
Maybe `Partial<CoreSetup>` would be better as it is a little less permissive?

Reply:
https://github.com/elastic/kibana/pull/39169#discussion_r299494525